src/model/table.py

```diff
@@ -25,27 +25,26 @@ def get_primary_key(self):
 
     # Will be implemented later in the project
     def __serialize_row__(self, data):
-        primary_key = self.get_primary_key()
+        primary_key = self. __get_row_primary_key__(data.get(self.get_primary_key()))
         can_overwrite = eval(self.__table_metadata__.overwrite)
-        path = Table.__get_row_path__(self.__path__, data.get(primary_key))
-        mode = 'w' if can_overwrite else 'x'
-        data[self.get_primary_key()] = os.path.basename(path).replace(".json","")
+        path = self.__get_row_path__(primary_key)
+        mode = 'w' if  can_overwrite else 'x'
```

Comment:
i would suggest to have enum here called FILE_MODE. 
can enum be characters, check this ?